cmake/FindMKL.cmake

```diff
@@ -82,13 +86,13 @@ if (ENABLE_MKLCPU_BACKEND OR ENABLE_MKLGPU_BACKEND)
   if(ENABLE_MKLCPU_THREAD_TBB)
     list(APPEND MKL_LINK_C ${TBB_LINK})
   endif()
-  if(ENABLE_MKLGPU_BACKEND)
+  if(USE_DPCPP_API)
```

Comment:
What is the purpose of using USE_DPCPP_API workaround?

Reply:
LAPACK's cpu backend links to libmkl_sycl because of the scratchpad functions not found in the c/fortran API. This workaround was suggested by @mkrainiuk to avoid impacting other domains.